### PR TITLE
Build JavaScript projects in parallel from PNPM

### LINKED ISSUE #35365: Run JavaScript builds in parallel from PNPM instead of Maven

### PR DESCRIPTION
**Pull request by jonkoops**
Closes #35365

### CODE REVIEW

**Review by stianst — COMMENTED** [review]
Shouldn't be built in `pom`, but rather within `js/pom`. Guess that's a bit of a challenge with `themes`?

**Comment by jonkoops** [comment]
Marking this as ready for review, @keycloak/ui @stianst can I get a round of reviews?

**Comment by jonkoops** [comment]
I did some quick benchmarking using `mvn install -DskipTests` from the project root with both Maven and PNPM caches primed (no dependencies downloaded). One run is a fresh workspace (as if you just cloned the repo), and the other a rebuild, here are the results:

**Fresh**
```
[INFO] Keycloak JavaScript Parent ......................... SUCCESS [ 20.941 s]
[INFO] Keycloak Admin UI .................................. SUCCESS [  0.167 s]
[INFO] Keycloak Account UI ................................ SUCCESS [  0.119 s]
[INFO] Keycloak Themes Vendor ............................. SUCCESS [  0.327 s]
...
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  01:33 min
[INFO] ------------------------------------------------------------------------
```

**Rebuild**
```
[INFO] Keycloak JavaScript Parent ......................... SUCCESS [  1.373 s]
[INFO] Keycloak Admin UI .................................. SUCCESS [  0.052 s]
[INFO] Keycloak Account UI ................................ SUCCESS [  0.032 s]
[INFO] Keycloak Themes Vendor ............................. SUCCESS [  0.122 s]
...
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  48.405 s
[INFO] ------------------------------------------------------------------------
```

And here are the same benchmarks performed on the `release/26.0` branch:

**Fresh**
```
[INFO] Keycloak JavaScript Parent ......................... SUCCESS [  0.004 s]
[INFO] Keycloak Admin UI .................................. SUCCESS [ 38.932 s]
[INFO] Keycloak Account UI ................................ SUCCESS [ 13.593 s]
[INFO] Keycloak Themes .................................... SUCCESS [  3.852 s]
...
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  02:07 min
[INFO] ------------------------------------------------------------------------
```

**Rebuild**
```
[INFO] Keycloak JavaScript Parent ......................... SUCCESS [  0.005 s]
[INFO] Keycloak Admin UI .................................. SUCCESS [ 28.890 s]
[INFO] Keycloak Account UI ................................ SUCCESS [ 13.457 s]
[INFO] Keycloak Themes .................................... SUCCESS [  3.656 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  01:34 min
[INFO] Finished at: 2024-12-04T20:13:08+01:00
[INFO] ------------------------------------------------------------------------
```

File: js/libs/keycloak-admin-client/pom.xml [hunk 1]
```diff
@@ -42,6 +42,15 @@
                         <groupId>com.github.eirslett</groupId>
                         <artifactId>frontend-maven-plugin</artifactId>
                         <executions>
+                            <execution>
```
Comment (stianst) [inline js/libs/keycloak-admin-client/pom.xml:45]: Why is a build needed in jboss-release profile?

File: js/libs/ui-shared/pom.xml [hunk 2]
```diff
@@ -42,6 +42,15 @@
                         <groupId>com.github.eirslett</groupId>
                         <artifactId>frontend-maven-plugin</artifactId>
                         <executions>
+                            <execution>
```
Comment (stianst) [inline js/libs/ui-shared/pom.xml:45]: Same here. Why is a build needed in jboss-release profile?

**Review by edewit — APPROVED** [review]

**Review by stianst — APPROVED** [review]

**Review by agagancarczyk — APPROVED** [review]

**Pull request merged**